dbms/src/Interpreters/Join.cpp

```diff
@@ -1075,6 +1089,11 @@ struct Adder<KIND, ASTTableJoin::Strictness::All, Map>
         }
 
         current_offset += rows_joined;
+        if (current_offset >= probe_process_info.max_block_size)
```

Comment:
In case of some unnecessary split, I think we can break here only if `current_offset >= std::max(probe_process_info.max_block_size, block.rows())`?

Reply:
> In case of some unnecessary split, I think we can break here only if `current_offset >= std::max(probe_process_info.max_block_size, block.rows())`?

done.   probe_process_info.max_block_size will be update to max(probe_process_info.max_block_size, block.rows()).